QMLFileType.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * QMLFileType.js - Represents a collection of QML files
+ *
+ * Copyright © 2020, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var log4js = require("log4js");
+var QMLFile = require("./QMLFile.js");
+var logger = log4js.getLogger("loctool.plugin.QMLFileType");
+
+var QMLFileType = function(project) {
+    this.type = "qml";
+    this.datatype = "qml";
+    this.resourceType = "ts";
+
+    this.project = project;
+    this.API = project.getAPI();
+
+    this.extensions = [ ".qml"];
+
+    this.extracted = this.API.newTranslationSet(project.getSourceLocale());
+    this.newres = this.API.newTranslationSet(project.getSourceLocale());
+    this.pseudo = this.API.newTranslationSet(project.getSourceLocale());
+};
+
+/**
+ * Return true if the given path is a qml or js file and is handled
+ * by the current file type.
+ *
+ * @param {String} pathName path to the file being questions
+ * @returns {boolean} true if the path is a qml file, or false
+ * otherwise
+ */
+QMLFileType.prototype.handles = function(pathName) {
+    logger.debug("QMLFileType handles " + pathName + "?");
+    var ret = false;
+    if ((pathName.length > 4 && pathName.substring(pathName.length - 4) === ".qml") ||
+        (pathName.length > 3 && pathName.substring(pathName.length - 3) === ".js")) {
```

Comment:
JS files can contain qml code?

Reply:
We have cases of using `qsTr` and `qsTranslate` in js file in QML apps. and it should be extracted that case and need to write to the ts file.